lib/src/flagged/flagged_bloc.dart

```diff
@@ -87,26 +92,33 @@ class FlaggedBloc extends Bloc<FlaggedEvent, FlaggedState> {
   void _registerListeners() {
       if (!_listenersRegistered) {
         _listenersRegistered = true;
-        repositoryStreamHandler = RepositoryEventStreamHandler(Type.publish, Event.msgsChanged, _updateMessages);
-        _messageListRepository.addListener(repositoryStreamHandler);
+        _repositoryStreamHandler = RepositoryEventStreamHandler(Type.publish, Event.msgsChanged, _updateMessages);
+        _messageListRepository.addListener(_repositoryStreamHandler);
       }
   }
 
   void _unregisterListeners() {
       if (_listenersRegistered) {
         _listenersRegistered = false;
-        _messageListRepository?.removeListener(repositoryStreamHandler);
+        _messageListRepository?.removeListener(_repositoryStreamHandler);
       }
   }
 
   void _updateMessages(Event event) => add(UpdateMessages());
 
   void _loadFlaggedMessages() async {
-    List<int> dateMakerIds = List();
-    Context context = Context();
-    List<int> messageIds = List.from(await context.getChatMessages(Chat.typeStarred, Context.chatListAddDayMarker));
+    final List<int> dateMakerIds = List();
+    final Context context = Context();
+    final List<int> messageIds = List.from(await context.getChatMessages(Chat.typeStarred, Context.chatListAddDayMarker));
+
+    if (null != _chatId) {
```

Comment:
Is it actually possible that the _chatId isNull?

Reply:
Yes! This is a new property I introduced for this ticket. It is only set (via `RequestFlaggedMessages` event) if I request flagged messages for the particular chat with **this** `chatId`. Otherwise we request flagged messages by giving a special `chatId` constat called [`Chat.typeStarred`](https://github.com/open-xchange/ox-coi/blob/OT-540_ListFlaggedMessagesPerConversation/lib/src/flagged/flagged_bloc.dart#L112).